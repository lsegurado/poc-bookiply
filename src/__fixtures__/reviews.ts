import { ReviewType } from "../types";

export const mockReviews: ReviewType[] = [
    {
        "headline": "Amazing Pool!",
        "comment": "Mauris justo ante, pulvinar eget consequat at, bibendum et lorem. Suspendisse lacus urna, fringilla sit amet commodo eget, condimentum et nisl. Pellentesque elit mi, porta in mi at, vulputate mattis lacus.",
        "author": "Alissa Stacey",
        "positiveFeedback": "The location is perfect",
        "negativeFeedback": null,
        "score": 4.1,
        "channel": "AIRBNB",
        "publishedAt": "2020-08-11T12:20:02.340Z",
        "id": "H5JlM3uFEF5X_ZU1Fn3iR"
    },
    {
        "headline": "Very nice host, and quite chill place.",
        "comment": "Nulla dictum ligula ac tortor egestas, sit amet ullamcorper risus sollicitudin. Morbi vitae leo accumsan, interdum ligula non, placerat ligula. Ut faucibus congue purus, vitae semper sapien viverra non. Pellentesque rhoncus porttitor diam, eu ultrices metus dictum at.",
        "author": "Alissa Stacey",
        "positiveFeedback": "The location is perfect",
        "negativeFeedback": "No parking spot.",
        "score": 4,
        "channel": "HOLIDU",
        "publishedAt": "2020-08-11T12:20:02.340Z",
        "id": "F4JJQJty9YU5nEp5Xb8gD"
    },
    {
        "headline": "Location is perfect!!",
        "comment": "Morbi porttitor nisl ipsum, a facilisis purus euismod eu. Praesent consequat interdum nisi ut auctor. Sed posuere est porta neque pretium viverra. Nulla vel finibus nulla.",
        "author": "Amirah Brandt",
        "positiveFeedback": "Everything you need is there",
        "negativeFeedback": null,
        "score": 4.1,
        "channel": "HOLIDU",
        "publishedAt": "2020-08-11T12:20:02.340Z",
        "id": "c3OBdkVzDnUMCJLfeTiiV"
    },
    {
        "headline": "Very nice host, and quite chill place.",
        "comment": "Donec lacus mi, tincidunt a pulvinar sed, malesuada sed nunc. Nullam euismod ultricies elit. Integer et magna et tortor viverra malesuada eget vitae eros. Mauris vitae ultricies mi.",
        "author": "Kaisha Melton",
        "positiveFeedback": "The location is perfect",
        "negativeFeedback": "A bit dusty",
        "score": 4,
        "channel": "BOOKINGCOM",
        "publishedAt": "2020-08-11T12:20:02.340Z",
        "id": "T6my_yaJZPHsFilSp9Uuc"
    },
    {
        "headline": "Very nice host, and quite chill place.",
        "comment": "Nulla dictum ligula ac tortor egestas, sit amet ullamcorper risus sollicitudin. Morbi vitae leo accumsan, interdum ligula non, placerat ligula. Ut faucibus congue purus, vitae semper sapien viverra non. Pellentesque rhoncus porttitor diam, eu ultrices metus dictum at.",
        "author": "Alissa Stacey",
        "positiveFeedback": null,
        "negativeFeedback": null,
        "score": 4.3,
        "channel": "HOLIDU",
        "publishedAt": "2020-08-11T12:20:02.340Z",
        "id": "-zCML8_Pw0CVABATSZpY5"
    },
    {
        "headline": "Lovely finca",
        "comment": "Phasellus id massa et ligula lobortis sollicitudin eu vel erat. Suspendisse faucibus aliquam dui, vel ultricies nulla fringilla nec. Nam mauris sapien, eleifend id finibus et, sollicitudin a velit.",
        "author": "Kaisha Melton",
        "positiveFeedback": "Yummy breakfast",
        "negativeFeedback": "Not the most central location.",
        "score": 4.2,
        "channel": "HOLIDU",
        "publishedAt": "2020-08-11T12:20:02.340Z",
        "id": "w6ZmU2DZDJso5YRFalkXN"
    },
    {
        "headline": "Amazing Pool!",
        "comment": "Donec lacus mi, tincidunt a pulvinar sed, malesuada sed nunc. Nullam euismod ultricies elit. Integer et magna et tortor viverra malesuada eget vitae eros. Mauris vitae ultricies mi.",
        "author": "Amirah Brandt",
        "positiveFeedback": "Everything you need is there",
        "negativeFeedback": null,
        "score": 4.3,
        "channel": "BOOKINGCOM",
        "publishedAt": "2020-08-11T12:20:02.340Z",
        "id": "xPz18IKGoOvIoRf8JL3eC"
    },
    {
        "headline": "Amazing Stay",
        "comment": "Mauris justo ante, pulvinar eget consequat at, bibendum et lorem. Suspendisse lacus urna, fringilla sit amet commodo eget, condimentum et nisl. Pellentesque elit mi, porta in mi at, vulputate mattis lacus.",
        "author": "Cadi Pham",
        "positiveFeedback": null,
        "negativeFeedback": "No parking spot.",
        "score": 4.5,
        "channel": "AIRBNB",
        "publishedAt": "2020-08-11T12:20:02.340Z",
        "id": "Z-QzdHfFygjiPgUITSW4l"
    },
    {
        "headline": "The standard chunk of Lorem Ipsum",
        "comment": "Donec lacus mi, tincidunt a pulvinar sed, malesuada sed nunc. Nullam euismod ultricies elit. Integer et magna et tortor viverra malesuada eget vitae eros. Mauris vitae ultricies mi.",
        "author": "Christopher Key",
        "positiveFeedback": null,
        "negativeFeedback": "AC was not working",
        "score": 4.4,
        "channel": "AIRBNB",
        "publishedAt": "2020-08-11T12:20:02.340Z",
        "id": "JEd63hyHnlbkbcALcbW2Q"
    },
    {
        "headline": "The standard chunk of Lorem Ipsum",
        "comment": "Donec ac est pulvinar massa ornare eleifend vel in magna. Aenean porttitor, nisi in tempor imperdiet, nisl ipsum sagittis eros, nec scelerisque sapien urna dictum quam.",
        "author": "Cadi Pham",
        "positiveFeedback": "Everything you need is there",
        "negativeFeedback": "We had to wait for 1h for the check-in.",
        "score": 4.4,
        "channel": "HOLIDU",
        "publishedAt": "2020-08-11T12:20:02.340Z",
        "id": "Axrhbze7AX1uI-KkJaWWg"
    },
    {
        "headline": "We had a blast",
        "comment": "Donec ac est pulvinar massa ornare eleifend vel in magna. Aenean porttitor, nisi in tempor imperdiet, nisl ipsum sagittis eros, nec scelerisque sapien urna dictum quam.",
        "author": "Jena Morgan",
        "positiveFeedback": "Super clean",
        "negativeFeedback": "Not the most central location.",
        "score": 4.2,
        "channel": "BOOKINGCOM",
        "publishedAt": "2020-08-11T12:20:02.340Z",
        "id": "3IwkYOXrm9PR3rM8ME9Ye"
    },
    {
        "headline": "Amazing Pool!",
        "comment": "Morbi porttitor nisl ipsum, a facilisis purus euismod eu. Praesent consequat interdum nisi ut auctor. Sed posuere est porta neque pretium viverra. Nulla vel finibus nulla.",
        "author": "Jena Morgan",
        "positiveFeedback": "Super clean",
        "negativeFeedback": null,
        "score": 4.3,
        "channel": "BOOKINGCOM",
        "publishedAt": "2020-08-11T12:20:02.340Z",
        "id": "5uyYAABQaPwVC7fCmBeG0"
    },
    {
        "headline": "Amazing Stay",
        "comment": "Mauris justo ante, pulvinar eget consequat at, bibendum et lorem. Suspendisse lacus urna, fringilla sit amet commodo eget, condimentum et nisl. Pellentesque elit mi, porta in mi at, vulputate mattis lacus.",
        "author": "Cadi Pham",
        "positiveFeedback": null,
        "negativeFeedback": null,
        "score": 4,
        "channel": "HOLIDU",
        "publishedAt": "2020-08-11T12:20:02.340Z",
        "id": "vGia4yqX1B999O9fISVQC"
    },
    {
        "headline": "Lovely finca",
        "comment": "Donec lacus mi, tincidunt a pulvinar sed, malesuada sed nunc. Nullam euismod ultricies elit. Integer et magna et tortor viverra malesuada eget vitae eros. Mauris vitae ultricies mi.",
        "author": "Molly Dodson",
        "positiveFeedback": "Host very welcoming",
        "negativeFeedback": "Not the most central location.",
        "score": 4.3,
        "channel": "AIRBNB",
        "publishedAt": "2020-08-11T12:20:02.340Z",
        "id": "15HoragWz4I5sYC90CX52"
    },
    {
        "headline": "Very nice host, and quite chill place.",
        "comment": "Aliquam vitae massa in orci auctor pharetra tincidunt vitae diam. Ut pretium accumsan risus quis mattis. Nulla interdum lacus a convallis commodo.",
        "author": "Cadi Pham",
        "positiveFeedback": "Host very welcoming",
        "negativeFeedback": "We had to wait for 1h for the check-in.",
        "score": 4.4,
        "channel": "BOOKINGCOM",
        "publishedAt": "2020-08-11T12:20:02.340Z",
        "id": "sptTRhW69u9_KFQGdEAo0"
    },
    {
        "headline": "The standard chunk of Lorem Ipsum",
        "comment": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Vestibulum lobortis, turpis eu molestie ultrices, velit lorem efficitur ante, sed malesuada libero neque ac lorem. Maecenas a mi vel magna placerat varius. Nam fringilla imperdiet nisl, id consectetur est auctor nec. Nunc molestie lacus odio, eget consectetur purus congue placerat.",
        "author": "Molly Dodson",
        "positiveFeedback": "The location is perfect",
        "negativeFeedback": null,
        "score": 4.5,
        "channel": "AIRBNB",
        "publishedAt": "2020-08-11T12:20:02.340Z",
        "id": "Jbxef7j8T-cUE2fT9jN5g"
    },
    {
        "headline": "Amazing Stay",
        "comment": "Donec ac est pulvinar massa ornare eleifend vel in magna. Aenean porttitor, nisi in tempor imperdiet, nisl ipsum sagittis eros, nec scelerisque sapien urna dictum quam.",
        "author": "Aneesha Conley",
        "positiveFeedback": "Yummy breakfast",
        "negativeFeedback": "AC was not working",
        "score": 4.5,
        "channel": "AIRBNB",
        "publishedAt": "2020-08-11T12:20:02.340Z",
        "id": "nuHTp9Gx0_hxP8oDsEqSM"
    },
    {
        "headline": "Amazing Stay",
        "comment": "Phasellus id massa et ligula lobortis sollicitudin eu vel erat. Suspendisse faucibus aliquam dui, vel ultricies nulla fringilla nec. Nam mauris sapien, eleifend id finibus et, sollicitudin a velit.",
        "author": "Amirah Brandt",
        "positiveFeedback": "Everything you need is there",
        "negativeFeedback": "We had to wait for 1h for the check-in.",
        "score": 4.4,
        "channel": "HOLIDU",
        "publishedAt": "2020-08-11T12:20:02.340Z",
        "id": "ss8eYKSUdBuDrp07XWEK-"
    },
    {
        "headline": "The standard chunk of Lorem Ipsum",
        "comment": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Vestibulum lobortis, turpis eu molestie ultrices, velit lorem efficitur ante, sed malesuada libero neque ac lorem. Maecenas a mi vel magna placerat varius. Nam fringilla imperdiet nisl, id consectetur est auctor nec. Nunc molestie lacus odio, eget consectetur purus congue placerat.",
        "author": "Findlay Mccarty",
        "positiveFeedback": "Host very welcoming",
        "negativeFeedback": "AC was not working",
        "score": 4.4,
        "channel": "HOLIDU",
        "publishedAt": "2020-08-11T12:20:02.340Z",
        "id": "YxH_fNbVEYExTb3UhinG0"
    },
    {
        "headline": "Amazing Stay",
        "comment": "Phasellus id massa et ligula lobortis sollicitudin eu vel erat. Suspendisse faucibus aliquam dui, vel ultricies nulla fringilla nec. Nam mauris sapien, eleifend id finibus et, sollicitudin a velit.",
        "author": "Cadi Pham",
        "positiveFeedback": "Everything you need is there",
        "negativeFeedback": null,
        "score": 4.5,
        "channel": "HOLIDU",
        "publishedAt": "2020-08-11T12:20:02.340Z",
        "id": "qiZsFAMNzBMw5ANCjnFPW"
    },
    {
        "headline": "Amazing Pool!",
        "comment": "Morbi porttitor nisl ipsum, a facilisis purus euismod eu. Praesent consequat interdum nisi ut auctor. Sed posuere est porta neque pretium viverra. Nulla vel finibus nulla.",
        "author": "Aneesha Conley",
        "positiveFeedback": "Yummy breakfast",
        "negativeFeedback": null,
        "score": 4.1,
        "channel": "AIRBNB",
        "publishedAt": "2020-08-11T12:20:02.340Z",
        "id": "st2dzKb6zSejMdsAm87H1"
    },
    {
        "headline": "Location is perfect!!",
        "comment": "Donec ac est pulvinar massa ornare eleifend vel in magna. Aenean porttitor, nisi in tempor imperdiet, nisl ipsum sagittis eros, nec scelerisque sapien urna dictum quam.",
        "author": "Kaisha Melton",
        "positiveFeedback": "Yummy breakfast",
        "negativeFeedback": "No parking spot.",
        "score": 4.1,
        "channel": "BOOKINGCOM",
        "publishedAt": "2020-08-11T12:20:02.340Z",
        "id": "ea3jm7RNPlr8h7NhfAq5a"
    },
    {
        "headline": "Lovely finca",
        "comment": "Mauris justo ante, pulvinar eget consequat at, bibendum et lorem. Suspendisse lacus urna, fringilla sit amet commodo eget, condimentum et nisl. Pellentesque elit mi, porta in mi at, vulputate mattis lacus.",
        "author": "Geraint Wagner",
        "positiveFeedback": null,
        "negativeFeedback": "A bit dusty",
        "score": 4.4,
        "channel": "AIRBNB",
        "publishedAt": "2020-08-11T12:20:02.340Z",
        "id": "ArRP9O2HWshpJmVBoOy6d"
    },
    {
        "headline": "Amazing Stay",
        "comment": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Vestibulum lobortis, turpis eu molestie ultrices, velit lorem efficitur ante, sed malesuada libero neque ac lorem. Maecenas a mi vel magna placerat varius. Nam fringilla imperdiet nisl, id consectetur est auctor nec. Nunc molestie lacus odio, eget consectetur purus congue placerat.",
        "author": "Christopher Key",
        "positiveFeedback": "Everything you need is there",
        "negativeFeedback": null,
        "score": 4.3,
        "channel": "AIRBNB",
        "publishedAt": "2020-08-11T12:20:02.340Z",
        "id": "Q1m93bEwmEuHyjeyPDite"
    },
    {
        "headline": "We had a blast",
        "comment": "Donec lacus mi, tincidunt a pulvinar sed, malesuada sed nunc. Nullam euismod ultricies elit. Integer et magna et tortor viverra malesuada eget vitae eros. Mauris vitae ultricies mi.",
        "author": "Jena Morgan",
        "positiveFeedback": "Super clean",
        "negativeFeedback": "A bit dusty",
        "score": 4.1,
        "channel": "HOLIDU",
        "publishedAt": "2020-08-11T12:20:02.340Z",
        "id": "_96evVZfhKjK6ZLlAQ8-4"
    },
    {
        "headline": "Very nice host, and quite chill place.",
        "comment": "Phasellus id massa et ligula lobortis sollicitudin eu vel erat. Suspendisse faucibus aliquam dui, vel ultricies nulla fringilla nec. Nam mauris sapien, eleifend id finibus et, sollicitudin a velit.",
        "author": "Molly Dodson",
        "positiveFeedback": "Super clean",
        "negativeFeedback": null,
        "score": 4.4,
        "channel": "HOLIDU",
        "publishedAt": "2020-08-11T12:20:02.340Z",
        "id": "S-r3ZCyDfZh1uIjHLpg83"
    },
    {
        "headline": "Amazing Stay",
        "comment": "Phasellus id massa et ligula lobortis sollicitudin eu vel erat. Suspendisse faucibus aliquam dui, vel ultricies nulla fringilla nec. Nam mauris sapien, eleifend id finibus et, sollicitudin a velit.",
        "author": "Molly Dodson",
        "positiveFeedback": null,
        "negativeFeedback": "A bit dusty",
        "score": 4.1,
        "channel": "BOOKINGCOM",
        "publishedAt": "2020-08-11T12:20:02.340Z",
        "id": "pYXq_WmTYGhTW4qL_CVb3"
    },
    {
        "headline": "Amazing Pool!",
        "comment": "Nulla dictum ligula ac tortor egestas, sit amet ullamcorper risus sollicitudin. Morbi vitae leo accumsan, interdum ligula non, placerat ligula. Ut faucibus congue purus, vitae semper sapien viverra non. Pellentesque rhoncus porttitor diam, eu ultrices metus dictum at.",
        "author": "Aneesha Conley",
        "positiveFeedback": "Everything you need is there",
        "negativeFeedback": "No parking spot.",
        "score": 4.4,
        "channel": "AIRBNB",
        "publishedAt": "2020-08-11T12:20:02.340Z",
        "id": "SexsgPkAdOIGcxF4BxMAo"
    },
    {
        "headline": "Lovely finca",
        "comment": "Phasellus id massa et ligula lobortis sollicitudin eu vel erat. Suspendisse faucibus aliquam dui, vel ultricies nulla fringilla nec. Nam mauris sapien, eleifend id finibus et, sollicitudin a velit.",
        "author": "Geraint Wagner",
        "positiveFeedback": "Host very welcoming",
        "negativeFeedback": "A bit dusty",
        "score": 4.4,
        "channel": "AIRBNB",
        "publishedAt": "2020-08-11T12:20:02.340Z",
        "id": "JfDtDxK7pFHpg1cW-_tbT"
    },
    {
        "headline": "Amazing Pool!",
        "comment": "Nulla dictum ligula ac tortor egestas, sit amet ullamcorper risus sollicitudin. Morbi vitae leo accumsan, interdum ligula non, placerat ligula. Ut faucibus congue purus, vitae semper sapien viverra non. Pellentesque rhoncus porttitor diam, eu ultrices metus dictum at.",
        "author": "Kaisha Melton",
        "positiveFeedback": null,
        "negativeFeedback": null,
        "score": 4.1,
        "channel": "AIRBNB",
        "publishedAt": "2020-08-11T12:20:02.340Z",
        "id": "7_qrD4rDCvMeQeBZYKQH-"
    }
]