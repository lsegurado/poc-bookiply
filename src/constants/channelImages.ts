import AIRBNB from '../assets/AIRBNB.svg';
import BOOKINGCOM from '../assets/BOOKINGCOM.svg';
import HOLIDU from '../assets/HOLIDU.svg';
import { ChannelType } from '../types';

export const channelImages: Record<ChannelType, string> = {
    AIRBNB,
    BOOKINGCOM,
    HOLIDU
};
