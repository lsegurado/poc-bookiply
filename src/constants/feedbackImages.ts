import thumbUp from '../assets/thumb-up.svg';
import thumbDown from '../assets/thumb-down.svg';

/**Images related to positive / negative feedback */
export const feedbackImages = {
    thumbUp,
    thumbDown
}