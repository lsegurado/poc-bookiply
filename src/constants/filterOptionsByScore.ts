/**Predefined options for filtering by score */
export const filterOptionsByScore = {
    /**Minimum score to look for */
    availableScoringFilters: [4.5, 4.0, 3.5, 3.0],
    /**Maximum score */
    maximumScore: 5,
    /**Precision used to send the filter by scores to the API */
    filterAccuracy: 0.1,
    /**Accuracy to be displayed by the stars in the filter */
    starsPrecision: 0.5
};
