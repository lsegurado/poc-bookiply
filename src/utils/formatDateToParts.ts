export const formatDateToParts = (date: string) => {
    const parsedDate = new Date(date);

    const [month, , day, , year] = Intl.DateTimeFormat('en', { day: 'numeric', month: 'long', year: 'numeric' }).formatToParts(parsedDate);

    return {
        month: month.value,
        day: day.value,
        year: year.value
    }
}