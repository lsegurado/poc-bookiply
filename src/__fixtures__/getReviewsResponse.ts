import { mockReviews } from "./reviews";

export const mockGetReviewsResponse = {
    value: mockReviews.map(({id, ...review}) => review),
    total: mockReviews.length
}