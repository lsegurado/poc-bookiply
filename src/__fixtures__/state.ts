import { RootState } from "../app/store";
import { initialState as initialStateReviews } from "../slices/reviewsSlice";
import { initialState as initialStateReviewsFilters } from "../slices/reviewsFiltersSlice";
import { mockReviews } from "./reviews";

export const idleState: RootState = {
    reviews: {
        reviews: mockReviews,
        status: 'idle',
        total: mockReviews.length
    },
    reviewsFilter: initialStateReviewsFilters
}
export const loadingState: RootState = {
    reviews: {
        ...initialStateReviews,
        status: 'loading'
    },
    reviewsFilter: initialStateReviewsFilters
}
export const failedState: RootState = {
    reviews: {
        ...initialStateReviews,
        status: 'failed'
    },
    reviewsFilter: initialStateReviewsFilters
}
export const idleWithNoResultsState: RootState = {
    reviews: initialStateReviews,
    reviewsFilter: initialStateReviewsFilters
}
export const mockState = jest.fn(() => idleState)