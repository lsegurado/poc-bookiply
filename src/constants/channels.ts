import { ChannelType } from "../types";
import { channelImages } from "./channelImages";

/**A channel list */
export const channels = Object.keys(channelImages) as ChannelType[]