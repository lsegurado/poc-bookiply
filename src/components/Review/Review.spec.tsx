import { render, RenderResult } from "@testing-library/react";
import { Review } from ".";
import { channelImages } from "../../constants/channelImages";
import { feedbackImages } from "../../constants/feedbackImages";
import { filterOptionsByScore } from "../../constants/filterOptionsByScore";
import { formatDateToParts } from "../../utils/formatDateToParts";
import { mockReviews } from "../../__fixtures__/reviews";

const review = mockReviews[1];
const {
    headline,
    comment,
    author,
    positiveFeedback,
    negativeFeedback,
    score,
    channel,
    publishedAt
} = review

describe('Review component', () => {
    let el: RenderResult;
    beforeEach(() => {
        el = render(<Review review={review} />);
    })
    it(`should match snapshot`, () => {
        expect(el).toMatchSnapshot();
    })
    it(`should render a score`, () => {
        expect(el.getByText(`${score.toFixed(1)} / ${filterOptionsByScore.maximumScore}`)).toBeDefined();
    })
    it(`should render a channel`, () => {
        expect(el.getAllByRole('img')[0].getAttribute('src')).toBe(channelImages[channel]);
    })
    it(`should render a headline`, () => {
        expect(el.getByText(headline)).toBeDefined();
    })
    it(`should render a comment`, () => {
        expect(el.getByText(comment)).toBeDefined();
    })
    it(`should render a positive feedback`, () => {
        expect(el.getAllByRole('img')[1].getAttribute('src')).toBe(feedbackImages.thumbUp);
        expect(el.getByText(positiveFeedback as string)).toBeDefined();
    })
    it(`should render a negative feedback`, () => {
        expect(el.getAllByRole('img')[2].getAttribute('src')).toBe(feedbackImages.thumbDown);
        expect(el.getByText(negativeFeedback as string)).toBeDefined();
    })
    it(`should render an author`, () => {
        expect(el.getByText(author)).toBeDefined();
    })
    it(`Should render a publish date`, () => {
        const { day, month, year } = formatDateToParts(publishedAt);
        expect(el.getByText(`Reviewed ${day} ${month} ${year}`)).toBeDefined();
    })
});