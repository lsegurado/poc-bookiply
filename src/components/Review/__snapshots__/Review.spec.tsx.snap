// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Review component should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <article
        class="article"
      >
        <section
          class="reviewSection"
        >
          <span
            class="css-173apo3"
          >
            4.0
             / 5
          </span>
          <img
            alt="HOLIDU"
            src="HOLIDU.svg"
          />
        </section>
        <h2
          class="headline"
        >
          Very nice host, and quite chill place.
        </h2>
        <p
          class="comment"
        >
          Nulla dictum ligula ac tortor egestas, sit amet ullamcorper risus sollicitudin. Morbi vitae leo accumsan, interdum ligula non, placerat ligula. Ut faucibus congue purus, vitae semper sapien viverra non. Pellentesque rhoncus porttitor diam, eu ultrices metus dictum at.
        </p>
        <p>
          <img
            alt="Positive feedback"
            class="css-sxswvr"
            src="thumb-up.svg"
          />
          The location is perfect
        </p>
        <p>
          <img
            alt="Negative feedback"
            class="css-sxswvr"
            src="thumb-down.svg"
          />
          No parking spot.
        </p>
        <p
          class="author"
        >
          Alissa Stacey
        </p>
        <span>
          Reviewed 
          11
           
          August
           
          2020
        </span>
      </article>
    </div>
  </body>,
  "container": <div>
    <article
      class="article"
    >
      <section
        class="reviewSection"
      >
        <span
          class="css-173apo3"
        >
          4.0
           / 5
        </span>
        <img
          alt="HOLIDU"
          src="HOLIDU.svg"
        />
      </section>
      <h2
        class="headline"
      >
        Very nice host, and quite chill place.
      </h2>
      <p
        class="comment"
      >
        Nulla dictum ligula ac tortor egestas, sit amet ullamcorper risus sollicitudin. Morbi vitae leo accumsan, interdum ligula non, placerat ligula. Ut faucibus congue purus, vitae semper sapien viverra non. Pellentesque rhoncus porttitor diam, eu ultrices metus dictum at.
      </p>
      <p>
        <img
          alt="Positive feedback"
          class="css-sxswvr"
          src="thumb-up.svg"
        />
        The location is perfect
      </p>
      <p>
        <img
          alt="Negative feedback"
          class="css-sxswvr"
          src="thumb-down.svg"
        />
        No parking spot.
      </p>
      <p
        class="author"
      >
        Alissa Stacey
      </p>
      <span>
        Reviewed 
        11
         
        August
         
        2020
      </span>
    </article>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Review component should match snapshote 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <article
        class="article"
      >
        <section
          class="reviewSection"
        >
          <span
            class="css-173apo3"
          >
            4.0
             / 5
          </span>
          <img
            alt="HOLIDU"
            src="HOLIDU.svg"
          />
        </section>
        <h2
          class="headline"
        >
          Very nice host, and quite chill place.
        </h2>
        <p
          class="comment"
        >
          Nulla dictum ligula ac tortor egestas, sit amet ullamcorper risus sollicitudin. Morbi vitae leo accumsan, interdum ligula non, placerat ligula. Ut faucibus congue purus, vitae semper sapien viverra non. Pellentesque rhoncus porttitor diam, eu ultrices metus dictum at.
        </p>
        <p>
          <img
            alt="Positive feedback"
            class="css-sxswvr"
            src="thumb-up.svg"
          />
          The location is perfect
        </p>
        <p>
          <img
            alt="Negative feedback"
            class="css-sxswvr"
            src="thumb-down.svg"
          />
          No parking spot.
        </p>
        <p
          class="author"
        >
          Alissa Stacey
        </p>
        <span>
          Reviewed 
          11
           
          August
           
          2020
        </span>
      </article>
    </div>
  </body>,
  "container": <div>
    <article
      class="article"
    >
      <section
        class="reviewSection"
      >
        <span
          class="css-173apo3"
        >
          4.0
           / 5
        </span>
        <img
          alt="HOLIDU"
          src="HOLIDU.svg"
        />
      </section>
      <h2
        class="headline"
      >
        Very nice host, and quite chill place.
      </h2>
      <p
        class="comment"
      >
        Nulla dictum ligula ac tortor egestas, sit amet ullamcorper risus sollicitudin. Morbi vitae leo accumsan, interdum ligula non, placerat ligula. Ut faucibus congue purus, vitae semper sapien viverra non. Pellentesque rhoncus porttitor diam, eu ultrices metus dictum at.
      </p>
      <p>
        <img
          alt="Positive feedback"
          class="css-sxswvr"
          src="thumb-up.svg"
        />
        The location is perfect
      </p>
      <p>
        <img
          alt="Negative feedback"
          class="css-sxswvr"
          src="thumb-down.svg"
        />
        No parking spot.
      </p>
      <p
        class="author"
      >
        Alissa Stacey
      </p>
      <span>
        Reviewed 
        11
         
        August
         
        2020
      </span>
    </article>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
